packages/web-frontend/src/selectors/measureSelectors.js

```diff
@@ -76,6 +77,18 @@ export const selectHasPolygonMeasure = createSelector(
   },
 );
 
+export const selectCurrentMeasureIds = createSelector([state => state], state => {
+  const currentMapOverlayIds = selectCurrentMapOverlayIds(state);
+  const { mapOverlayHierarchy } = state.mapOverlayBar;
+  const hierarchyMapOverlays = flattenMapOverlayHierarchy(mapOverlayHierarchy);
+
+  return hierarchyMapOverlays.map(hierarchyMapOverlay =>
+    currentMapOverlayIds.includes(hierarchyMapOverlay.mapOverlayId)
+      ? hierarchyMapOverlay.measureIds
+      : [],
+  );
```

Comment:
This reminds me of `selectDisplayedMeasureIds`... The difference is that `selectDisplayedMeasureIds` uses `.flat()` in the end. Perhaps we don't need it theere?

Reply:
That's a bit confusing but they are slightly different. `selectCurrentMeasureIds` is what have been selected in checkbox while `selectDisplayedMeasureIds` is what have been switch on.

A particular example can be `selectCurrentMeasureIds` = [[measure1, measure2], [measure3]] `selectDisplayedMeasureIds` = [measure1, measure2], 

But I feel that we can filter the result of `selectCurrentMeasureIds` by displayed map overlay ids. Hope that make more sense?

Reply:
I feel this part of logic can be simpler when we  are working on double date picker because there will be two responses which is separated by map overlay id. At that time we don't need to group measure ids here in reselector.